Source/AwsCommonRuntimeKit/mqtt/Mqtt5Packets.swift

```diff
@@ -220,22 +220,74 @@ public class PublishPacket: CStruct {
                                 raw_publish_view.user_properties =
                                     UnsafePointer<aws_mqtt5_user_property>(_userPropertyPointer)
                             }
-                            return withOptionalByteCursorPointerFromString(
+                            return withOptionalByteCursorPointerFromStrings(
                             responseTopic,
-                            correlationData,
-                            contentType) { cResponseTopic, cCorrelationData, cContentType in
+                            contentType) { cResponseTopic, cContentType in
                                 raw_publish_view.content_type = cContentType
-                                raw_publish_view.correlation_data = cCorrelationData
                                 raw_publish_view.response_topic = cResponseTopic
 
-                                return body(raw_publish_view)
+                                return withOptionalByteCursorPointerFromData(to: self.correlationData) {  cCorrelationData in
+                                    if let _cCorrelationData = cCorrelationData {
+                                        raw_publish_view.correlation_data = _cCorrelationData
+                                    }
+                                    return body(raw_publish_view)
+                                }
                             }
                         }
                     }
                 }
             }
         }
     }
+
+    static func convertFromNative(_ from: UnsafePointer<aws_mqtt5_packet_publish_view>?) -> PublishPacket? {
+        if let _from = from {
```

Comment:
Same comments as the other one. 
1. Guard will be better
2. If you make this a failable init?, you won't need to make different variables to pass to the init function, you can assign it directly like s`elf.publishView = `.

Reply:
Failable init is a good idea. As we have a list of PRs chained after this one, such big update would affect the following PRs. I will have a separate PR to make all of the "convertFromNative" to failable init.